python3/unittest/test_usb_scan.py

```diff
@@ -107,65 +104,70 @@ def test_log(m):
 
 
 class TestUsbScan(unittest.TestCase):
-
     def setUp(self):
-        try:
-            self.work_dir = tempfile.mkdtemp(prefix="test_usb_scan")
-        except:
-            raise
+        self.work_dir = tempfile.mkdtemp(prefix="test_usb_scan")
 
     def tearDown(self):
         shutil.rmtree(self.work_dir, ignore_errors=True)
 
     @nottest
-    def test_usb_common(self, moc_devices, moc_interfaces, moc_results,
-                        path="./scripts/usb-policy.conf"):
-        import usb_scan
+    def test_usb_common(
+        self, moc_devices, moc_interfaces, moc_results, path="./scripts/usb-policy.conf"
+    ):
+        usb_scan = get_module("usb_scan", "../libexec/usb_scan.py")
+
         mock_setup(usb_scan, moc_devices, moc_interfaces, path)
 
         devices, interfaces = usb_scan.get_usb_info()
 
+        usb_scan.log_list(devices)
+        usb_scan.log_list(interfaces)
+
         pusbs = usb_scan.make_pusbs_list(devices, interfaces)
 
         # pass pusbs in json to XAPI
         self.assertEqual(sorted(pusbs), sorted(moc_results))
 
     @nottest
-    def test_usb_exit(self, devices, interfaces, results,
-                      path="./scripts/usb-policy.conf", msg=""):
+    def test_usb_exit(
+            self, devices, interfaces, results,
+            path="./scripts/usb-policy.conf",
+            msg=""
+            ):  # pylint: disable=too-many-arguments
         with self.assertRaises(SystemExit) as cm:
             self.test_usb_common(devices, interfaces, results, path)
         if msg:
-            self.assertIn(msg, cm.exception.code)
+            # cm.exception.code is int type whose format
+            # looks like "duplicated tag'vid' found,
+            # malformed line ALLOW:vid=056a vid=0314 class=03"
+            self.assertIn(msg, cm.exception.code)  # pytype: disable=wrong-arg-types
```

Comment:
This is inside an `if msg:` so we don't actually know from the code if this assertIn() really runs.
But if the CodeCov line I see in the reivew is not fake, I see that this line in the test is executed (which is why it is a good idea to have code coverage for tests).

In https://github.com/xapi-project/xen-api/pull/5424/commits/57fffaa19b711258355adfd78b9f4c75ce7e9d44#r1497078077 @gangj wrote:
> I think it would be better to resolve this pylint warning.

Maybe this is a language barrier issue, but "resolve" means to fix, and this warning is now disabled, not fixed.

It's not obvious to me that is going on as wrong-arg-types would mean that the args don't have the same type, so something should be better explained ere.
The comment says that cm.exception.code is an `is int type whose format looks like`, but that is likely just very confusing language is normally `int` would refer to an `integer`. Instead, it seems like this wants to say that exception code is really the string that is quoted afterwards.

But as can be seen also, this is not actually a test function but a helper function, that is called by multiple tests, so the code coverage just says that one test covers this line, not all.

That means to clear up this confusion, it would be great to split move the code of this test function to the locations where it was called, and then it will be actually shown where `msg` is checked correctly.

As it stands, an `if msg` in a test function called by other functions is not a good idea, as it does not tell which tests check the msg. It's better to move the `assertIn()` to those locations that passed `msg` to the function, and then the test will be clearer.

There are some things that should not be done in a test, like looping over a list whose contents are not known. Such test code can result in no execution of the checks in the loop. Such guidelines should be adopted and implemented here.